pkg/kubelet/gpu/nvidia/nvidia_gpu_manager.go

```diff
@@ -152,6 +152,12 @@ func (ngm *nvidiaGPUManager) AllocateGPU(pod *v1.Pod, container *v1.Container) (
 			return nil, fmt.Errorf("Failed to allocate GPUs because of issues with updating GPUs in use: %v", err)
 		}
 	}
+	// Check if GPUs have already been allocated. If so return them right away.
+	// This can happen if a container restarts for example.
+	if devices := ngm.allocated.getGPUs(string(pod.UID), container.Name); devices != nil {
+		glog.V(2).Infof("Found pre-allocated GPUs for container %q in Pod %q: %v", container.Name, pod.UID, devices.List())
+		return devices.List(), nil
```

Comment:
Just to make sure I understand this: AllocateGPU is only called when either none of the GPUs for a container have been allocated, or all of the GPUs for a container have been allocated.  There would never arise a situation where we find 3 GPUs already allocated, but require 4, right?